packages/core/src/synthesis/Synthesizer.ts

```diff
@@ -399,9 +463,10 @@ export class Synthesizer {
       )
     );
     // add autolabel statement
-    this.updateProg(autoLabel(this.currentProg));
+    // TODO: find out what to label
+    // this.updateProg(autoLabel(this.currentProg));
     return {
-      prog: this.currentProg,
+      prog: sortStmts(dedupStmts(this.currentProg)), // sort and deduplicate statements to make sure edits don't introduce compiler errors
```

Comment:
What does this mean exactly? Is the idea that object declarations always get sorted before their usages in constructors/functions/predicates?